### PR TITLE
Feature/d3 asim 3179: Restoring results endpoint from last known state

### PR DESCRIPTION
**Pull request by faizan2590**

### CODE REVIEW

**Review by Hassan754 — APPROVED** [review]

File: d3a_interface/sim_results/bills.py [hunk 1]
```diff
@@ -45,11 +45,23 @@ def cumulative_bills(self):
                 "spent_total": round_floats_for_ui(results['spent_total']),
                 "earned": round_floats_for_ui(results['earned']),
                 "penalties": round_floats_for_ui(results['penalties']),
+                "penalty_energy": round_floats_for_ui(results['penalty_energy']),
                 "total": round_floats_for_ui(results['total'])
             }
             for uuid, results in self.cumulative_bills_results.items()
         }
 
+    def restore_state_from_area(self, area_uuid, last_known_state_data):
```
Comment (hannesdiedrich) [inline d3a_interface/sim_results/bills.py:54]: Here the function name is misleading IMO. You dont restore from the area, but from the DB.
How about: `restore_area_results_state` ?

File: d3a_interface/sim_results/cumulative_grid_trades.py [hunk 2]
```diff
@@ -210,28 +220,6 @@ def _accumulate_area_trades(cls, area, parent, flattened_area_core_stats_dict,
...
-        if parent is not None:
-            results[area['uuid']].append(CumulativeGridTrades._external_trade_entries(
-                area, accumulated_trades))
-
-        for child in area['children']:
-            results = cls.generate_cumulative_grid_trades_for_all_areas(
-                accumulated_trades, child, area, results
-            )
-        return results
-
```
Comment (hannesdiedrich) [inline d3a_interface/sim_results/cumulative_grid_trades.py:234]: Nice, so now we save this effort. It is now done in `generate_cumulative_grid_trades_target_area` only for the target area on the d3a-web side, right?

File: d3a_interface/sim_results/bills.py (hunk 1 above)
Comment (faizan2590) [reply d3a_interface/sim_results/bills.py]: https://github.com/gridsingularity/d3a-interface/pull/153/commits/1e8fda3fb5c818e1c0c4b139820914df7243cb87

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by Hassan754 — APPROVED** [review]
LGTM

**Pull request merged**